File: 2021_spring/coding_meetup_02/regional_team_names/team_names.cpp
/// @author Benjamin Alcocer

#include <iostream>
#include <string>

using namespace std;

int main()
{
    string input;
    getline(cin, input);

    int index = input.find_first_of("-");

    // Another way: (we just want the index of the first '-')
    /*
    for (int i = 0; i < input.size(); ++i)
    {
        if (input[i] == '-')
        {
            index = i;
            break;
        }
    }
    */

    int teamSize = input.size() - (index + 1);
    if (index > 1 && index <= 8 && teamSize >= 1 && teamSize <= 24)
    {
        cout << "YES";
    }
    else
    {
        cout << "NO";
    }

    return 0;
}

File: 2021_spring/coding_meetup_02/regional_team_names/team_names.py
# @author Benjamin Alcocer

array = input().split('-', 1)
print('YES' if len(array) == 2 and len(array[0]) > 1 and len(array[0]) <= 8 and len(array[1]) >= 1 and len(array[1]) <= 24 else 'NO')
